src/algebra/ordered_monoid.lean

```diff
@@ -1163,4 +1208,21 @@ instance [ordered_cancel_comm_monoid α] : ordered_cancel_add_comm_monoid (addit
   ..additive.add_left_cancel_semigroup,
   ..additive.ordered_add_comm_monoid }
 
+instance [linear_ordered_add_comm_monoid α] : linear_ordered_comm_monoid (multiplicative α) :=
+{ ..multiplicative.linear_order,
+  ..multiplicative.ordered_comm_monoid }
+
+instance [linear_ordered_comm_monoid α] : linear_ordered_add_comm_monoid (additive α) :=
+{ ..additive.linear_order,
+  ..additive.ordered_add_comm_monoid }
+
+instance [linear_ordered_add_comm_monoid_with_top α] :
+  linear_ordered_comm_monoid_with_zero (multiplicative (order_dual α)) :=
+{ zero := (⊤ : α),
+  zero_mul := top_add,
+  mul_zero := add_top,
+  zero_le_one := (le_top : (0 : α) ≤ ⊤),
+  ..multiplicative.ordered_comm_monoid,
+  ..multiplicative.linear_order }
```

Comment:
This one should be in the same file as `linear_ordered_add_comm_monoid_with_top`, whether that ends up being this file or the other one.